File: ZenCare.Services/Database/WellnessService.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using ZenCare.Model.Enums;

namespace ZenCare.Services.Database;

public class WellnessService
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(150)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(1000)]
    public string? Description { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal Price { get; set; }

    public int DurationMinutes { get; set; }

    public ServiceStatus Status { get; set; } = ServiceStatus.Draft;

    public int ServiceCategoryId { get; set; }

    [ForeignKey(nameof(ServiceCategoryId))]
    public ServiceCategory ServiceCategory { get; set; } = null!;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    public ICollection<EmployeeService> EmployeeServices { get; set; } = new List<EmployeeService>();
}

File: ZenCare.Mobile/lib/models/wellness_service.dart
class WellnessService {
  WellnessService({
    required this.id,
    required this.name,
    this.description,
    required this.durationMinutes,
    required this.price,
    required this.serviceCategoryId,
    required this.serviceCategoryName,
    required this.isActive,
  });

  final int id;
  final String name;
  final String? description;
  final int durationMinutes;
  final double price;
  final int serviceCategoryId;
  final String serviceCategoryName;
  final bool isActive;

  factory WellnessService.fromJson(Map<String, dynamic> json) {
    return WellnessService(
      id: json['id'] as int? ?? 0,
      name: json['name'] as String? ?? '',
      description: json['description'] as String?,
      durationMinutes: json['durationMinutes'] as int? ?? 0,
      price: (json['price'] as num?)?.toDouble() ?? 0,
      serviceCategoryId: json['serviceCategoryId'] as int? ?? 0,
      serviceCategoryName: json['serviceCategoryName'] as String? ?? '',
      isActive: json['isActive'] as bool? ?? false,
    );
  }
}
